app/assets/javascripts/map.js

```diff
@@ -80,39 +100,101 @@ class Map {
       componentRestrictions: {country: 'ca'}
     };
 
-    const input = document.getElementById('address');
+    const input = document.getElementById('addressInput');
     const autocomplete = new google.maps.places.Autocomplete(input, options);
+    // TODO execute search when autocomplete option has been selected doesnt work
+    // autocomplete.addListener('place_changed', this.handleSearchSubmit);
   }
 
   addListeners() {
-    document.getElementById('AddressSearchModal').addEventListener('submit', this.handleSearchSubmit);
+    document.getElementById('addressSearchBar').addEventListener('keyup', this.enableFindButton)
+    document.getElementById('addressSearchBar').addEventListener('submit', this.handleSearchSubmit);
     this.drawingManager.addListener('polygoncomplete', this.handlePolygonCreated);
     this.removeLastControl.addEventListener('click', this.handleRemoveLastPolygon);
     this.removeAllControl.addEventListener('click', this.handleRemoveAllPolygons);
+    document.getElementById('doneSelectingArea').addEventListener('click', this.handleDoneSelecting);
+    document.getElementsByName("serviceExpeditionCost").forEach((element) => { 
+      element.addEventListener('click', this.handleExpeditionInfoClick); 
+    });
+    document.getElementById("addBag").addEventListener('click', this.handleAddBagClick);
+    document.getElementById("removeBag").addEventListener('click', this.handleRemoveBagClick);
+    document.getElementById("doneAddOns").addEventListener('click', this.handleDoneAddOns);
+    document.getElementById("quote_email").addEventListener('keyup', () => this.enableSubmitButton);
+    document.getElementById("quote_phone_number").addEventListener('keyup', () => this.enableSubmitButton);
```

Comment:
Something is funky here... you're passing an inline anonymous function that returns the function itself... Why isn't this just `addEventListener('keyup', this.enableSubmitButton)`? (note that `enableSubmitButton` must be bound, so you can do `addEventListener('keyup', this.enableSubmitButton.bind(this)`). Could also look into event delegation (like commented on L118, and have the form listen to all keyup events in the children... somewhat contrary to my comment on L110, but in that case it's a single input field in the form, not many, like here)